src/template_parser.h

```diff
@@ -138,12 +138,33 @@ struct ParserTraits<wchar_t> : public ParserTraitsBase<>
     {
         auto tmpStr = str.substr(range.startOffset, range.size());
         std::string result;
-        result.resize(tmpStr.size(), '\0');
+        const wchar_t *srcPtr;
+        size_t srcSize, destSize;
+
+        srcPtr = tmpStr.c_str();
+        srcSize = tmpStr.size();
 #ifdef _MSC_VER
-        size_t dummy = 0;
-        wcstombs_s(&dummy, &result[0], result.size(), tmpStr.c_str(), tmpStr.size());
+        destSize = 0;
+        wcstombs_s(&destSize, NULL, 0, srcPtr, srcSize);
+        if (destSize == 0) {
+            return "";
+        }
+        result.resize(destSize);
+        wcstombs_s(&destSize, &result[0], destSize, srcPtr, srcSize);
+        if (destSize == 0) {
+            return "";
+        }
+        result.resize(destSize-1);
 #else
-        wcstombs(&result[0], tmpStr.c_str(), result.size());
+        destSize = wcstombs(NULL, srcPtr, 0);
+        if (destSize == (size_t)-1) {
```

Comment:
It's better to use `numeric_limits<size_t>::max()` here in order to avoid like 'signed to unsigned cast'. Something like:
```c++
constexpr auto max_size = numeric_limits<size_t>::max();
// ... use max_size as a constant ...
```

Reply:
It is definitely better to use StringConverter<std::wstring, std::string>::DoConvert since it reuses existing code. However StringConverter<std::wstring, std::string>::DoConvert has a couple of bugs too:

1. The "++ len;" on line 75 is unnecessary because wcsrtombs_s (unlike wcsrtombs) already counts the terminating NULL character towards the required buffer size in len.

2. The result variable is not shrunk after the second call wcsrtombs_s or wcsrtombs. So result ends up holding an extra terminating NULL which should not really be a part of std::string. Due to the first bug, strings converted by wcsrtombs_s end up with two extra characters, one for the terminating NULL and another garbage byte due to the extra byte allocated initially.

I can provide a patch using StringConverter<std::wstring, std::string>::DoConvert, or I can let you make the changes yourself, since it is a small modification.

Reply:
Yes, it would be better to fix StringConverter implementation, if you could. Thanks!